### ISSUE TITLE
"IF NOT EXISTS" clause has no effect when creating a stream

### ISSUE THREAD

**Issue by naamancurtis** [confluentinc/ksql#7074]
**Describe the bug**
When creating a stream with the KSQL-DB REST API and with the `IF NOT EXISTS` clause, an error response is returned in the case that the stream already exists. The [docs](https://docs.ksqldb.io/en/latest/developer-guide/ksqldb-reference/create-stream/) state that the statement won't fail if this clause is used. 

**To Reproduce**
Steps to reproduce the behavior, include:
1. The version of KSQL: v0.15.0
2. Sample source data - N/a
3. Any SQL statements you ran:
```sql
CREATE STREAM IF NOT EXISTS MY_STREAM (
   id VARCHAR KEY
) WITH (
    kafka_topic = 'my-topic',
    partitions = 1,
    replicas = 1,
    value_format = 'AVRO'
);
```

**Expected behavior**
As the stream already existed, I would have expected some kind of success response, or at least something that does not include the `error_code` property, or define the `@type` as "statement_error"

**Actual behaviour**
-  When sending the request, I get the following response structure:
```txt
[{
    "@type":  "statement_error",
    "error_code":  40001,
    "message":  "Cannot add stream \'MY_STREAM\': A stream with the same name already exists",
    "statementText": "..."
}]
```

**Additional context**
The created stream is using the confluent Schema Registry for it's schema.

**Comment by spena** [comment]
This was fixed in 0.14 - https://github.com/confluentinc/ksql/issues/6050

I just tested in 0.15 docker and latest master, but I don't see the error you are experiencing.

```
curl -X "POST" "http://localhost:8088/ksql" \
>      -H "Accept: application/vnd.ksql.v1+json" \
>      -d $'{
>   "ksql": "create stream if not exists s1(id int) with (kafka_topic=\'s1\', value_format=\'json\', partitions=1);",
>   "streamsProperties": {}
> }'
[{"@type":"currentStatus","statementText":"CREATE STREAM IF NOT EXISTS S1 (ID INTEGER) WITH (KAFKA_TOPIC='s1', KEY_FORMAT='KAFKA', PARTITIONS=1, VALUE_FORMAT='JSON');","commandId":"stream/`S1`/create","commandStatus":{"status":"SUCCESS","message":"Cannot add stream `S1`: A stream with the same name already exists.","queryId":null},"commandSequenceNumber":6,"warnings":[]}]
```

The message the stream does not exist is still sent, but it is not an error.

**Comment by spena** [comment]
I'll close this as fixed.
